cmd/soroban-cli/src/config/mod.rs

```diff
@@ -1,114 +1,102 @@
 use std::path::PathBuf;
 
 use clap::{arg, command};
+use secret::StellarSigner;
 use serde::{Deserialize, Serialize};
 
-use soroban_rpc::Client;
+use crate::signer::{self, Stellar};
+use crate::xdr::{Transaction, TransactionEnvelope};
+use crate::Pwd;
 
-use crate::{
-    signer,
-    xdr::{Transaction, TransactionEnvelope},
-    Pwd,
-};
-
-use self::{network::Network, secret::Secret};
+use self::network::Network;
 
 pub mod alias;
 pub mod data;
 pub mod locator;
 pub mod network;
 pub mod secret;
+pub mod sign_with;
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     #[error(transparent)]
-    Network(#[from] network::Error),
+    SignWith(#[from] sign_with::Error),
     #[error(transparent)]
     Secret(#[from] secret::Error),
     #[error(transparent)]
-    Config(#[from] locator::Error),
+    Network(#[from] network::Error),
     #[error(transparent)]
-    Rpc(#[from] soroban_rpc::Error),
+    Locator(#[from] locator::Error),
     #[error(transparent)]
     Signer(#[from] signer::Error),
 }
 
 #[derive(Debug, clap::Args, Clone, Default)]
 #[group(skip)]
 pub struct Args {
-    #[command(flatten)]
-    pub network: network::Args,
-
     #[arg(long, visible_alias = "source", env = "STELLAR_ACCOUNT")]
     /// Account that signs the final transaction. Alias `source`. Can be an identity (--source alice), a secret key (--source SC36…), or a seed phrase (--source "kite urban…").
     pub source_account: String,
 
-    #[arg(long)]
-    /// If using a seed phrase, which hierarchical deterministic path to use, e.g. `m/44'/148'/{hd_path}`. Example: `--hd-path 1`. Default: `0`
-    pub hd_path: Option<usize>,
-
     #[command(flatten)]
-    pub locator: locator::Args,
+    pub sign_with: sign_with::Args,
 }
 
 impl Args {
-    pub fn key_pair(&self) -> Result<ed25519_dalek::SigningKey, Error> {
-        let key = self.account(&self.source_account)?;
-        Ok(key.key_pair(self.hd_path)?)
+    pub fn signer(&self) -> Result<StellarSigner, Error> {
+        let (account, prompt) = self.sign_with.sign_with_key.as_ref().map_or_else(
+            || (&self.source_account, false),
+            |s| (s, !self.sign_with.yes),
+        );
+        Ok(self
+            .sign_with
+            .locator
+            .account(account)?
+            .signer(self.sign_with.hd_path, prompt)?)
     }
 
-    pub async fn sign_with_local_key(&self, tx: Transaction) -> Result<TransactionEnvelope, Error> {
-        self.sign(tx).await
+    pub async fn public_key(&self) -> Result<stellar_strkey::ed25519::PublicKey, Error> {
+        Ok(self.signer()?.get_public_key().await?)
     }
 
-    #[allow(clippy::unused_async)]
     pub async fn sign(&self, tx: Transaction) -> Result<TransactionEnvelope, Error> {
-        let key = self.key_pair()?;
-        let Network {
-            network_passphrase, ..
-        } = &self.get_network()?;
-        Ok(signer::sign_tx(&key, &tx, network_passphrase)?)
+        Ok(self
```

Comment:
I think `sign(&self, tx: Transaction)` should live in `StellarSigner` (or I guess to be more precise the `Stellar` trait)

Reply:
As mentioned before we need the network information which `sign_with` has. So we can't move this into the trait since that interface requires the network passphrase.

Reply:
Resolving this conversation, as it can be changed later if needed